src/components/ImageGeneration/Field.tsx

```diff
@@ -0,0 +1,265 @@
+/* eslint-disable jsx-a11y/alt-text */
+/* eslint-disable @next/next/no-img-element */
+import type { ExtendedChain, Token } from '@lifi/sdk';
+import { AvatarBadgeSSR } from '../AvatarBadge/SSR/AvatarBadgeSSR';
+import { FieldSkeleton } from './FieldSkeleton';
+import type { ImageTheme } from './ImageGeneration.types';
+
+function formatDecimal(n: number): string | number {
+  // Check if the number is a whole number
+  if (Number.isInteger(n)) {
+    return n; // Return the number without decimals
+  }
+
+  // Convert to string to check the number of decimals
+  const decimalPart = n.toString().split('.')[1];
+
+  // If it has 2 or fewer decimal places, return it with 2 decimals
+  if (decimalPart && decimalPart.length <= 2) {
+    return n.toFixed(2);
+  }
+
+  // Otherwise, return the number rounded to 6 decimal places
+  return n.toFixed(6);
+}
+
+const Field = ({
+  sx,
+  token,
+  chain,
+  type,
+  amount = 0,
+  extendedHeight,
+  theme,
+  amountUSD,
+  routeAmount,
+  routeAmountUSD,
+  highlighted,
+  fullWidth,
+  showSkeletons,
+}: {
+  sx?: any; //SxProps<Theme>;
+  token?: Token | null;
+  chain?: ExtendedChain | null;
+  theme?: ImageTheme;
+  type:
+    | 'amount'
+    | 'token'
+    | 'quote'
+    | 'review'
+    | 'quote-amount'
+    | 'amount-selection'
+    | 'button'
+    | 'title'
+    | 'card-title'
+    | 'success';
+  amount?: number | null;
+  amountUSD?: number | null;
+  highlighted?: boolean | null;
+  routeAmount?: number | null;
+  routeAmountUSD?: number | null;
+  extendedHeight?: boolean;
+  fullWidth?: boolean;
+  showSkeletons?: boolean;
+}) => {
+  // Function to calculate top offset based on conditions
+  const getOffset = () => {
+    if (type === 'amount') {
+      return 46;
+    }
+    if (type === 'quote') {
+      if (extendedHeight) {
+        return 56;
+      }
+      return 16;
+    } else {
+      return 46;
+    }
+  };
+  const getWidth = () => {
+    if (type === 'quote') {
+      return 315;
+    } else if (fullWidth) {
+      return 368;
+    } else {
+      return 174;
+    }
+  };
+
+  const containerOffset = getOffset();
+  const containerWidth = getWidth();
+
+  return (
+    <div style={{ display: 'flex', width: containerWidth }}>
+      <div
+        style={{
+          display: 'flex',
+          height: extendedHeight ? 149.6 : type === 'quote' ? 112 : 104,
+          borderRadius: '12px',
+          borderWidth: '1px',
+          justifyContent: 'space-between',
+          borderStyle: 'solid',
+          // borderColor: '#E5E1EB',
+          ...(highlighted && {
+            boxShadow: `inset 0 0 0 2px ${theme === 'dark' ? '#653BA3' : '#31007A'}`,
+          }),
+          ...(type === 'quote' && {
+            flexDirection: 'column',
+            justifyContent: 'flex-start',
+          }),
+          padding: `${containerOffset}px 16px 16px`,
+          ...sx,
+        }}
+      >
+        {type !== 'button' && (
```

Comment:
Would it make sense to split this into separate files, given that there’s an if statement in most cases? It could help improve maintainability.

Reply:
That was a great idea! Lots of work to clean up the mess I left, but finally we have seperate components for different fields which will be much easier to maintain! Thanks for the input!